app/code/community/Bolt/Boltpay/controllers/ApiController.php

```diff
@@ -54,7 +54,9 @@ public function hookAction()
                 $this->handleDiscountHook();
                 return;
             }
-
+            if (!empty($reference)) {
```

Comment:
This lookup does not appear to be necessary and has been excluded for performance.  Moreover, it is duplicate code. 
 The call for this would be executed here:

https://github.com/BoltApp/bolt-magento1/blob/595e499dc4f612adb4ebd95c1b80b94ded615cc0/app/code/community/Bolt/Boltpay/controllers/ApiController.php#L63-L68